File: frontend/Haocean-Mobile/api/sync.js
import request from '@/utils/request'

// 检查数据更新
export function checkSync(query) {
  return request({
    url: '/system/sync/check',
    method: 'get',
    params: query
  })
}

// 同步行业分类
export function syncIndustries() {
  return request({
    url: '/system/sync/industries',
    method: 'get'
  })
}

// 同步文书模板
export function syncTemplates(industryId) {
  return request({
    url: '/system/sync/templates',
    method: 'get',
    params: { industryId }
  })
}

// 上报执法记录
export function syncRecords(data) {
  return request({
    url: '/system/sync/records',
    method: 'post',
    data: data
  })
}

// 上报监管单位
export function syncSubjects(data) {
  return request({
    url: '/system/sync/subjects',
    method: 'post',
    data: data
  })
}

// 获取同步状态
export function getSyncStatus(query) {
  return request({
    url: '/system/sync/status',
    method: 'get',
    params: query
  })
}

// 重试同步
export function retrySync(data) {
  return request({
    url: '/system/sync/retry',
    method: 'post',
    data: data
  })
}

File: app/admin/router/system/sync.go
package system

import (
	"github.com/gin-gonic/gin"
	"haocean/health-enforcement/app/admin/api/system"
	"haocean/health-enforcement/app/core/utils/jwt"
)

// InitSync 初始化数据同步路由
func InitSync(e *gin.Engine) {
	v := e.Group("system")
	{
		v.GET("/sync/check", system.CheckSync)
		v.GET("/sync/industries", system.SyncIndustries)
		v.GET("/sync/templates", system.SyncTemplates)

		auth := v.Group("")
		auth.Use(jwt.JWTAuthMiddleware())
		{
			auth.POST("/sync/records", system.SyncRecords)
			auth.POST("/sync/subjects", system.SyncSubjects)
			auth.GET("/sync/status", system.GetSyncStatus)
			auth.POST("/sync/retry", system.RetrySync)
			auth.GET("/sync/list", system.ListSync)
		}
	}
}
